src/cromshell/submit/command.py

```diff
@@ -0,0 +1,131 @@
+                                files=submission_params)
+
+    # Check to make sure that we actually submitted the job correctly
+    # 1. Check for any initial failure by server to accept the job.
+    http_utils.check_http_request_status_code(request_out)
+
+    # 2. Check messages from server for workflow problems.
+    # Get our workflow status and job ID:
+    workflow_status = json.loads(request_out.content)
+
+    # 2.A If the status is not `Submitted`, something went wrong: Todo: did a manual test but not one with failed workflow
+    if workflow_status['status'] != "Submitted":
+        log.display_logo(io_utils.dead_turtle)
+        LOGGER.error("Error: Server reports that job was not properly submitted.")
+        LOGGER.error(f"Server response:{request_out.text}")
+        raise Exception("Error: Server reports that job was not properly submitted.\n"
+            f"Server response:{request_out.text}")
+
+    # 2.B If the ID is not an ID, something went wrong:
+    if io_utils.validate_workflow_id(workflow_status['id']) != 0:
+        log.display_logo(io_utils.dead_turtle)
+        LOGGER.error("Error: Did not get a valid ID back. Something went wrong.")
+        LOGGER.error(f"Server response:{request_out.text}")
+        raise Exception("Error: Did not get a valid ID back. Something went wrong.\n"
+            f"Server response:{request_out.text}")
+
+    # If we get here, we successfully submitted the job and should track it locally:
+    log.display_logo(io_utils.turtle)
+    io_utils.pretty_print_json(request_out.text)
+
+    # Create a directory to hold function input files, using server name
+    run_directory = Path(config.config_dir)/config.folder_url/workflow_status['id']
+    Path.mkdir(Path(run_directory), parents=True, exist_ok=True)  # should it be ok that it exists?
+
+    # Copy input to run directory
+    shutil.copy(wdl, run_directory)
+    shutil.copy(wdl_json, run_directory)
+    if options_json is not None:
+        shutil.copy(options_json, run_directory)
+    if dependencies_zip is not None:
+        shutil.copy(dependencies_zip, run_directory)
+
+    # Update config.submission_file:
+    submission_row = [datetime.now().strftime("%Y%m%d_%H%M%S"),
+                      config.cromwell_server, workflow_status['id'],
+                      PurePath(wdl).name, workflow_status['status']]
+
+    with open(config.submission_file, 'a') as f:
+        writer = csv.writer(f, delimiter='\t', lineterminator="\n")
+        writer.writerow(submission_row)
+
+    return 0
+
+
+def womtool_validate_wdl_and_json(wdl: str, wdl_json: str):
+    """If womtool is found in PATH, validates wdl and json"""
+
+    womtool_path = shutil.which("womtool")
+    if womtool_path is not None:
+        validation_output = subprocess.run([womtool_path, 'validate', wdl, '-i',
+                                            wdl_json],
+                                           capture_output=True)
+        if validation_output.returncode == 0:
```

Comment:
Just want to share a trick from the C world:

to avoid accidentally assign a value to a variable, follow the convention of 
`if 0 == var`, this way if one accidentally writes `if 0 = var`, the computer barks.

It's a habit thing.

Reply:
done